src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package com.facebook.openwifirrm.ucentral.models;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association;
+import com.facebook.openwifirrm.ucentral.models.State.Interface.SSID.Association.Rate;
+
+public class AggregatedState {
+
+	public class AggregatedRate {
+		public long bitrate;
+		public int chwidth;
+		public List<Integer> mcs;
+
+		/** Constructor with no args */
+		public AggregatedRate() {
+			this.mcs = new ArrayList<>();
+		}
+
+		/** Constructor */
+		public AggregatedRate(List<Rate> rates) {
+			int size = rates.size();
+			if (size == 0) {
+				return;
+			}
+			add(rates);
+		}
+
+		/** Add a list of Rates to the AggregatedRate */
+		public void add(List<Rate> rates) {
+			if (rates.isEmpty()) {
```

Comment:
You do not need this check, the code will just not enter the for loop body if there are no elements.